src/TraceEvent/TraceEventSession.cs

```diff
@@ -733,9 +740,14 @@ public unsafe bool EnableKernelProvider(KernelTraceEventParser.Keywords flags, K
                 Marshal.ThrowExceptionForHR(TraceEventNativeMethods.GetHRFromWin32(dwErr));
                 m_IsActive = true;
 
-                if (OperatingSystemVersion.AtLeast(62) && StackCompression)
+                if (StackCompression && OperatingSystemVersion.AtLeast(OperatingSystemVersion.Win10))
                 {
-                    ETWControl.EnableStackCaching(m_SessionHandle.DangerousGetHandle());
+                    var info = new TraceEventNativeMethods.TRACE_STACK_CACHING_INFO{ Enabled = 1 };
+                    TraceEventNativeMethods.TraceSetInformation(
+                        m_SessionHandle.DangerousGetHandle(),
```

Comment:
We should add an overload of TraceSetInformation that takes a safe handle to "hide" the call to DangerousGetHandle

Reply:
Looks like that already exists.

Reply:
Actually, that overload (taking a SafeTraceHandle) already exists. Can we use it instead?
The call to TraceSetInformation a little higher up that passes zero for the TRACEHANDLE, we can allow `null` there? I think the "null" case is why the "dangerous" version of TraceSetInformation was not made private.